src/storage/src/hummock/local_version_manager.rs

```diff
@@ -406,50 +405,57 @@ impl LocalVersionManager {
         }
     }
 
-    pub fn build_shared_buffer_item_batches(
-        kv_pairs: Vec<(Bytes, StorageValue)>,
-        epoch: HummockEpoch,
-    ) -> Vec<SharedBufferItem> {
-        kv_pairs
-            .into_iter()
-            .map(|(key, value)| {
-                (
-                    Bytes::from(FullKey::from_user_key(key.to_vec(), epoch).into_inner()),
-                    value.into(),
-                )
-            })
-            .collect_vec()
-    }
-
-    pub async fn write_shared_buffer(
+    pub fn build_shared_buffer_batch(
         &self,
         epoch: HummockEpoch,
         compaction_group_id: CompactionGroupId,
         kv_pairs: Vec<(Bytes, StorageValue)>,
         table_id: u32,
-    ) -> HummockResult<usize> {
-        let sorted_items = Self::build_shared_buffer_item_batches(kv_pairs, epoch);
-        let batch = SharedBufferBatch::new(
+    ) -> SharedBufferBatch {
+        let sorted_items = build_shared_buffer_item_batches(kv_pairs, epoch);
+        SharedBufferBatch::new(
             sorted_items,
             epoch,
             self.buffer_tracker.buffer_event_sender.clone(),
             compaction_group_id,
             table_id,
-        );
+        )
+    }
+
+    pub fn write_shared_buffer_batch(
+        &self,
+        batch: SharedBufferBatch,
+    ) -> HummockResult<Option<oneshot::Receiver<()>>> {
```

Comment:
We use such `rx.await` to support blocking write. So it's better to do the `rx.await` inside the `write_shared_buffer_batch` instead of returning the `rx`.